### ISSUE TITLE
Adding a zero-order polynomial fails

### ISSUE THREAD

**Issue by thomasaarholt** [hyperspy/hyperspy#1602]
Adding a constant background to a model by adding a zero-order polynomial background fails when trying to look at it:

```python
m = s.create_model(auto_background=False)
m.add_polynomial_background(0)
m[-1]
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/Users/thomas/miniconda3/envs/hs/lib/python3.5/site-packages/IPython/core/formatters.py in __call__(self, obj)
    670                 type_pprinters=self.type_printers,
    671                 deferred_pprinters=self.deferred_printers)
--> 672             printer.pretty(obj)
    673             printer.flush()
    674             return stream.getvalue()

/Users/thomas/miniconda3/envs/hs/lib/python3.5/site-packages/IPython/lib/pretty.py in pretty(self, obj)
    381                             if callable(meth):
    382                                 return meth(obj, self, cycle)
--> 383             return _default_pprint(obj, self, cycle)
    384         finally:
    385             self.end_group()

/Users/thomas/miniconda3/envs/hs/lib/python3.5/site-packages/IPython/lib/pretty.py in _default_pprint(obj, p, cycle)
    501     if _safe_getattr(klass, '__repr__', None) not in _baseclass_reprs:
    502         # A user-provided repr. Find newlines and replace them with p.break_()
--> 503         _repr_pprint(obj, p, cycle)
    504         return
    505     p.begin_group(1, '<')

/Users/thomas/miniconda3/envs/hs/lib/python3.5/site-packages/IPython/lib/pretty.py in _repr_pprint(obj, p, cycle)
    699     """A pprint that just redirects to the normal repr function."""
    700     # Find newlines and replace them with p.break_()
--> 701     output = repr(obj)
    702     for idx,output_line in enumerate(output.splitlines()):
    703         if idx:

/Users/thomas/hyperspy/hyperspy/_components/polynomial.py in __repr__(self)
     65     def __repr__(self):
     66         text = "%s order Polynomial component" % ordinal(
---> 67             self.get_polynomial_order())
     68         if self.name:
     69             text = "%s (%s)" % (self.name, text)

/Users/thomas/hyperspy/hyperspy/_components/polynomial.py in get_polynomial_order(self)
     48 
     49     def get_polynomial_order(self):
---> 50         return len(self.coefficients.value) - 1
     51 
     52     def function(self, x):

TypeError: object of type 'float' has no len()

```

**Comment by thomasaarholt** [comment]
I think the coefficient value isn't forced to be a tuple when initialised.

**Comment by ericpre** [comment]
Fixed in #1989.
